packages/blade/src/components/Box/BaseBox/storybookArgTypes.ts

```diff
@@ -0,0 +1,242 @@
+  descriptionLength?: 'long';
+} = {}): StorybookArgTypes<StyledProps> => {
+  const commonStyledPropsProperties: {
+    table: { category?: 'StyledProps' | null };
+    control: { type: string };
+  } = {
+    ...commonProperties,
+    table: {
+      category,
+    },
+  };
+
+  const restStyledPropsArgTypes = Object.fromEntries(
+    Object.entries(defaultStyledPropsObject)
+      .filter(([key]) => !key.includes('margin'))
+      .map(([key, _value]) => {
+        const cssPropertyName = key.replace(/[A-Z]/g, (m) => `-${m.toLowerCase()}`);
+        return [
+          key,
+          {
+            ...commonProperties,
+            description: `**CSS property \`${cssPropertyName}\`** ${
+              descriptionLength === 'long' ? styledPropsSupportMessage : ''
+            }\n\n<a target="_blank" href="https://developer.mozilla.org/en-US/docs/Web/CSS/${cssPropertyName}">MDN Docs for ${cssPropertyName}</a><br/><br/>`,
+            table: {
+              category,
+              type: {
+                summary: `MakeValueResponsive<CSSObject['${key}']>`,
+              },
+            },
+          },
+        ];
+      }),
+  );
+
+  return {
+    margin: {
+      ...commonStyledPropsProperties,
+      description: `**Margin Shorthand**${
+        descriptionLength === 'long' ? styledPropsSupportMessage : ''
+      }\n\n${descriptionLength === 'long' ? spacingTypeShorthandDescription : ''}`,
+    },
+    marginX: {
+      ...commonStyledPropsProperties,
+      description: `**Horizontal Margin**${
+        descriptionLength === 'long' ? styledPropsSupportMessage : ''
+      }\n\n${descriptionLength === 'long' ? spacingTypeDescription : ''}`,
+    },
+    marginY: {
+      ...commonStyledPropsProperties,
+      description: `**Vertical Margin**${
+        descriptionLength === 'long' ? styledPropsSupportMessage : ''
+      }\n\n${descriptionLength === 'long' ? spacingTypeDescription : ''}`,
+    },
+    marginTop: {
+      ...commonStyledPropsProperties,
+      description: `**CSS Property \`margin-top\`**${
+        descriptionLength === 'long' ? styledPropsSupportMessage : ''
+      }\n\n Supports same values as marginX, and marginY`,
+    },
+    marginRight: {
+      ...commonStyledPropsProperties,
+      description: `**CSS Property \`margin-right\`**${
+        descriptionLength === 'long' ? styledPropsSupportMessage : ''
+      }\n\n Supports same values as marginX, and marginY`,
+    },
+    marginBottom: {
+      ...commonStyledPropsProperties,
+      description: `**CSS Property \`margin-bottom\`**${
+        descriptionLength === 'long' ? styledPropsSupportMessage : ''
+      }\n\n Supports same values as marginX, and marginY`,
+    },
+    marginLeft: {
+      ...commonStyledPropsProperties,
+      description: `**CSS Property \`margin-left\`**${
+        descriptionLength === 'long' ? styledPropsSupportMessage : ''
+      }\n\n Supports same values as marginX, and marginY`,
+    },
+    ...restStyledPropsArgTypes,
+  };
+};
```

Comment:
Can you also add some sub headings to the table to categorize the properties like we did in rfc props table? flex, grid, etc.

Reply:
We will have to manually define each prop we add and add it to correct category then. A bit of pain on maintenance end. Also the props are sorted according to how we added in RFC so flex props will be one after the other.